Sources/ApolloCodegenLib/ApolloSchemaOptions.swift

```diff
@@ -9,8 +9,48 @@ public struct ApolloSchemaOptions {
     case schemaDefinitionLanguage = "graphql"
   }
   
-  let apiKey: String?
-  let endpointURL: URL
+  /// How to attempt to download your schema
+  public enum DownloadMethod: Equatable {
+
+    case registry(_ settings: RegistrySettings)
+    ///   - endpointURL: The endpoint to hit to download your schema.
+    case introspection(endpointURL: URL)
+    
+    public struct RegistrySettings: Equatable {
+      public let apiKey: String
+      public let graphID: String
+      public let variant: String?
+      
+      /// Designated initializer
+      ///
+      /// - Parameters:
+      ///   - apiKey: The API key to use when retrieving your schema.
+      ///   - graphID: The identifier of the graph to fetch. Can be found in Apollo Studio.
```

Comment:
There's nothing explicitly named "graph id" in Apollo Studio that I can see, but I'm guessing it's the dynamic part of the URL (after "/graph/"). If so, this is also the part between "service:" and the key itself in the token.

I've changed the name of our graph in Apollo Studio, but the name in the URL and token retains the old name so I'm guessing the "graph id" is the full thing and not just the id-looking part after the old name. I'll give it a shot and see what works!

Reply:
My thoughts above were accurate, and it works 👍 An internet-savvy person can probably figure it out, but it's not entirely obvious what to put as the `graphID`.

Reply:
Yep, that's being addressed on the Studio end shortly.